python/ray/_raylet.pyx

```diff
@@ -673,7 +673,9 @@ cdef increase_recursion_limit():
         int new_limit = current_limit * 2
         cdef extern from *:
             """
-#if PY_VERSION_HEX >= 0x30B00A4
+#if PY_VERSION_HEX >= 0x30C0000
```

Comment:
take from https://github.com/ray-project/ray/pull/45653, thanks so much @pcmoritz for the heavy lifting work